rotkehlchen/api/rest.py

```diff
@@ -946,6 +946,29 @@ def user_change_password(
         else:
             return api_response(OK_RESULT, status_code=HTTPStatus.OK)
 
+    @require_premium_user(active_check=False)
+    def user_premium_key_remove(self, name: str) -> Response:
+        """Returns successful result if API keys are successfully removed"""
+        result_dict: Dict[str, Any] = {'result': None, 'message': ''}
+        success: bool
+
+        if name != self.rotkehlchen.data.username:
+            result_dict['message'] = f'Provided user "{name}" is not the logged in user'
+            return api_response(result_dict, status_code=HTTPStatus.BAD_REQUEST)
+
```

Comment:
Can you move this logic also into the rotkehlchen object? I would like to have the logic of DB and backend session removal of premium in one function. You can still keep the `deactivate_premium_status()` function too since who knows we may need to use such a thing at some point.

Reply:
A bit confused. Do you want me to move this into the `deactivate_premium_status()` func or a separate func (within rotkehlchen) that calls the `deactivate_premium_status()` func?

Reply:
> a separate func (within rotkehlchen) that calls the deactivate_premium_status() func?

this

Reply:
Done.